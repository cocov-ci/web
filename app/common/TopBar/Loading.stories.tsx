import type { Meta, StoryObj } from '@storybook/react'

import TopBarLoading from './Loading'

const meta: Meta<typeof TopBarLoading> = {
  title: 'Common/TopBar/Loading',
  component: TopBarLoading,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof TopBarLoading>

export const Loading: Story = {}
