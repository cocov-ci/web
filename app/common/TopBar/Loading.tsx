import Loading from '../Loading'

import styles from './TopBar.module.scss'

const LoadingToolbar = () => {
  return (
    <div className={styles.topBar}>
      <Loading
        alignment="left"
        count={1}
        height="65px"
        type="skeleton"
        width="300px"
      />
      <Loading
        alignment="right"
        count={1}
        height="50px"
        type="skeleton"
        width="400px"
      />
    </div>
  )
}

export default LoadingToolbar
