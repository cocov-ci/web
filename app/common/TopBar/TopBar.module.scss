@use "sass:map";
@import "styles/variables.module";

.topBar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: map.get($topbarSize, "height");
  margin-bottom: map.get($topbarSize, "marginbottom");

  .title {
    font-size: 38px;
    letter-spacing: -0.9px;
    font-variation-settings: "wght" 700;
  }

  .description {
    font-size: $fontDescription;
    letter-spacing: 0;
    font-variation-settings: "wght" 400;
    padding-left: 8px;
  }

  .actions {
    flex: 1;
  }
}
