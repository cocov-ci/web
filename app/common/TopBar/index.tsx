import classNames from 'classnames'
import React from 'react'

import styles from './TopBar.module.scss'

interface TopBarProps {
  children?: React.ReactNode
  className?: string
  title?: string
  description?: string
}

const TopBar = ({ title, description, children, className }: TopBarProps) => {
  return (
    <div className={classNames(styles.topBar, className)}>
      <div>
        <h1 className={styles.title}>{title}</h1>
        <p className={styles.description}>{description}</p>
      </div>
      <div>{children}</div>
    </div>
  )
}

export default TopBar
