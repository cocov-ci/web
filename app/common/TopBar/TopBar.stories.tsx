import type { Meta, StoryObj } from '@storybook/react'

import TopBar from './index'

const meta: Meta<typeof TopBar> = {
  title: 'Common/TopBar',
  component: TopBar,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof TopBar>

export const Default: Story = {
  args: {
    title: 'Title here...',
    description: 'Description here...',
  },
}
