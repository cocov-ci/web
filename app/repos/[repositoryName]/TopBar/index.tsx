'use client'

import Button from 'app/common/Button'
import PillNav from 'app/common/PillNav'
import TopBar from 'app/common/TopBar'
import { useSegments } from 'context/SegmentsContext'
import { RepositoryProps } from 'types/Repositories'

type index = 'summary' | 'badges' | 'settings' | undefined

interface NavButtonProps {
  index: index
  activeItem: index
  href: string
  text: string
}

const NavButton = ({ activeItem, index, text, href }: NavButtonProps) => {
  const isActive = activeItem === index

  return (
    <Button
      href={isActive ? undefined : href}
      style={isActive ? 'secondary' : 'inactive'}
    >
      {text}
    </Button>
  )
}

const TopBarComponent = ({
  name,
  default_branch,
  description,
}: RepositoryProps) => {
  const segments = useSegments()

  let activeItem: index

  if (segments.length >= 3 && segments[2] === 'branches') {
    activeItem = 'summary'
  } else if (segments.length === 3 && segments[2] === 'badges') {
    activeItem = 'badges'
  } else if (segments.length >= 3 && segments[2] === 'settings') {
    activeItem = 'settings'
  }

  return (
    <TopBar description={description} id="topBar" title={name}>
      <PillNav>
        <NavButton
          activeItem={activeItem}
          href={
            default_branch
              ? `/repos/${name}/branches/${default_branch}`
              : `/repos/${name}`
          }
          index="summary"
          text="Summary"
        />
        <NavButton
          activeItem={activeItem}
          href={`/repos/${name}/badges`}
          index="badges"
          text="Badges"
        />
        <NavButton
          activeItem={activeItem}
          href={`/repos/${name}/settings`}
          index="settings"
          text="Settings"
        />
      </PillNav>
    </TopBar>
  )
}

export default TopBarComponent
