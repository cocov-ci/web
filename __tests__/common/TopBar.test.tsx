import { render, screen } from '@testing-library/react'

import TopBar from 'app/common/TopBar'

import '@testing-library/jest-dom'

describe('TopBar', () => {
  it('renders TopBar', () => {
    render(
      <TopBar description="description" title="title">
        <p>children</p>
      </TopBar>,
    )

    expect(screen.getByText('description')).toBeVisible()
    expect(screen.getByText('title')).toBeVisible()
    expect(screen.getByText('children')).toBeVisible()
  })

  it('renders TopBar snapshots', () => {
    const { container } = render(<TopBar />)
    expect(container).toMatchSnapshot()
  })
})
