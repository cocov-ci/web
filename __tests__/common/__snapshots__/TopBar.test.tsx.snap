// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/TopBar renders TopBar snapshots 1`] = `
<div>
  <div
    class="topBar"
  >
    <div>
      <h1
        class="title"
      />
      <p
        class="description"
      />
    </div>
    <div />
  </div>
</div>
`;
